src/metrics.jl

```diff
@@ -180,20 +215,20 @@ end
             @simd for I in eachindex(a, b)
                 ai = a[I]
                 bi = b[I]
-                s = eval_reduce(d, s, eval_op(d, ai, bi))
+                s = eval_reduce(d, s, p === nothing ? eval_op(d, ai, bi) : eval_op(d, ai, bi, p[I]))
             end
         else
             for (Ia, Ib) in zip(eachindex(a), eachindex(b))
                 ai = a[Ia]
                 bi = b[Ib]
-                s = eval_reduce(d, s, eval_op(d, ai, bi))
+                s = eval_reduce(d, s, p === nothing ? eval_op(d, ai, bi) : eval_op(d, ai, bi, p[Ia]))
             end
         end
     end
     return eval_end(d, s)
 end
 result_type(dist::UnionMetrics, ::AbstractArray{T1}, ::AbstractArray{T2}) where {T1, T2} =
-    typeof(eval_end(dist, eval_op(dist, one(T1), one(T2))))
+    typeof(parameters(dist) === nothing ? eval_end(dist, eval_op(dist, one(T1), one(T2))) : evaluate(dist, one(T1), one(T2)))
```

Comment:
Why not call `eval_end` where there are parameters? Also, add a line break after `:`.